src/client/datascience/editor-integration/hoverProvider.ts

```diff
@@ -94,13 +101,19 @@ export class HoverProvider implements INotebookExecutionLogger, vscode.HoverProv
                     const notebooks = this.getMatchingNotebooks(document);
                     if (notebooks && notebooks.length) {
                         // Just use the first one to reply if more than one.
-                        const match = await Promise.race(
-                            notebooks.map((n) => this.variableProvider.getMatchingVariable(word, n, t))
+                        const attributes = await Promise.race(
+                            notebooks.map((n) => this.variableProvider.getVariableProperties!(word, n, t))
```

Comment:
I might toss on a comment here about the use of !. Had to actually look it up myself to see if this was safe. It's a bit funny since it's optional on IJupyterVariables, but this class now demands that it exists. It works since we are specifically injecting the kernelVariables, but a comment explaining that here might be nice.